core/src/main/java/org/bitcoinj/core/Transaction.java

```diff
@@ -1731,12 +1730,14 @@ public void checkCoinBaseHeight(final int height)
     /** Loops the outputs of a coinbase transaction to locate the witness commitment. */
     public Sha256Hash findWitnessCommitment() {
         checkState(isCoinBase());
-        for (TransactionOutput out : Lists.reverse(outputs)) {
-            Script scriptPubKey = out.getScriptPubKey();
-            if (ScriptPattern.isWitnessCommitment(scriptPubKey))
-                return ScriptPattern.extractWitnessCommitmentHash(scriptPubKey);
-        }
-        return null;
+        List<TransactionOutput> reversed = new ArrayList<>(outputs);
+        Collections.reverse(reversed);
```

Comment:
It may not matter much, but isn't there a performance difference here? It reverses the array physically, rather than just iterating backwards as before.

Does the stream API have a "reverse stream"?

Reply:
You should be able to use `return Lists.reverse(outputs).stream()…`.

Reply:
Yes, there is probably a slight performance difference.  However, there are only a small number of outputs in a transaction so copying and reversing the list will still be fast. Also `findWitnessCommitment()` is only used for testing.

> Does the stream API have a "reverseStream()"?

No. You can sort with a reverse comparator, but that won't work in this case.

> You should be able to use return Lists.reverse(outputs).stream()….

`Lists` is a Guava class.